src/extension.ts

```diff
@@ -3,38 +3,57 @@ import * as vscode from 'vscode';
 import WebSocket from 'ws';
 import QuickPickItem = vscode.QuickPickItem;
 import QuickPickOptions = vscode.QuickPickOptions;
-import { fixRemoteUrl, getURL } from './utils';
+import * as utils from './utils';
 
 export function activate(context: vscode.ExtensionContext) {
+    context.subscriptions.push(vscode.commands.registerCommand('devtools-for-chrome.attach', async () => {
+        attach(context);
+    }));
 
-    context.subscriptions.push(vscode.commands.registerCommand('devtools-for-chrome.start', async () => {
-        const settings = vscode.workspace.getConfiguration('vscode-devtools-for-chrome');
-        const hostname = settings.get('hostname') as string || 'localhost';
-        const port = settings.get('port') as number || 9222;
+    context.subscriptions.push(vscode.commands.registerCommand('devtools-for-chrome.launch', async () => {
+        launch(context);
+    }));
+}
 
-        const checkDiscoveryEndpoint = (url: string) => {
-            return getURL(url, { headers: { Host: 'localhost' } });
-        };
+const address = 'localhost';
+const port = 9222;
 
-        const jsonResponse = await checkDiscoveryEndpoint(`http://${hostname}:${port}/json/list`)
-            .catch(() => checkDiscoveryEndpoint(`http://${hostname}:${port}/json`));
+async function launch(context: vscode.ExtensionContext) {
+    const portFree = await utils.isPortFree(address, port);
+    if (portFree) {
+        utils.launchLocalChrome('about:blank');
+    }
 
-        const responseArray = JSON.parse(jsonResponse);
-        if (Array.isArray(responseArray)) {
-            const items: QuickPickItem[] = [];
+    attach(context);
```

Comment:
Do you still want to attach if the port isn't free? Should it be picking a different port automatically, or would that just be confusing.

Reply:
Also, if the settings hostname had been changed to something other than localhost, calling attach might not actually enumerate the targets on the chrome instance that was just launched. Perhaps you need to pass in the hostname and port to attach when you are calling it here?

Reply:
I was thinking that it would just re-attach rather than re-launch. The idea being people can just reuse the launch command after the first command.